ivadomed/transforms.py

```diff
@@ -914,3 +952,21 @@ def apply_preprocessing_transforms(transforms, seg_pair, roi_pair=None):
             'gt_metadata': metadata_roi
         }
     return (seg_pair, roi_pair)
+
+
+def preprare_transforms(transform_dict, requires_undo=True):
+    """
+    This function seperates the preprocessing transforms from the others and generates the undo transforms related.
+    :param transform_dict (dict): Dictionary containing the transforms and there parameters
+    :param requires_undo (bool): Boolean indicating if transforms can be undone
+    :return: transform lst containing the preprocessing transforms and regular transforms, UndoCompose object containing
+    the transform to undo
+    """
+    training_undo_transform = None
+    if requires_undo:
+        training_undo_transform = UndoCompose(Compose(transform_dict.copy()))
```

Comment:
For the undo, we do not need to split prepro vs non_prepro because thery are all done only once, correct?

Reply:
Exactly :) we still want to undo all the transforms